src/cortexm.rs

```diff
@@ -46,8 +46,10 @@ pub fn subroutine_eq(addr1: u32, addr2: u32) -> bool {
 /// The contents of the vector table
 #[derive(Debug)]
 pub struct VectorTable {
-    // entry 0
+    // entry 0: Initial SP
     pub initial_stack_pointer: u32,
+    // entry 1: Reset vector
+    pub reset: u32,
```

Comment:
There was some discussion if we can extract the reset vector from the elf or not and how to make this as stable as possible. Please see https://github.com/knurling-rs/probe-run/pull/383#issuecomment-1438840061. Do you think we can always™️ extract the reset vector the way you implemented it?

Reply:
From the linked discussion:

> It could be that the Flash memory at 0x2000_0000 is aliased at address 0x0000_0000 but I have not checked.

I think that's a typo: the Flash memory is (on STM32s for example; in principe it could be anywhere the instruction fetches are legal) at 0x0800_0000 and the SRAM is (always) at 0x2000_0000.

However, that aside, I looked it up in the ARMv7-M ARM and it seems that I misremembered some details and a much more reliable way would be to read the IVT address from the VTOR register on attach and then use that. I can implement that later.

![Screenshot_20230622_171704](https://github.com/knurling-rs/probe-run/assets/54771/4a43b848-c2c1-45c6-a559-7d45b7089749)

Reply:
> I can implement that later.

That would be nice. Maybe in a follow up PR.

In any case I we will need to test the current PR with the rp pico, because that one seems to behave a bit differently than many other microcontrollers because of the bootloader.

Reply:
I can do the testing. Just need to set it up.